httpsync/message.go

```diff
@@ -0,0 +1,96 @@
+package httpsync
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+)
+
+// HeadMsg represents the head cid from the src.
+type HeadMsg struct {
+	CidStr string
+}
+
+// SignedHeadMsgEnvelope is the signed envelope of a HeadMsg. It includes the
+// public key of the signer so the receiver can verify it and convert it to a
+// peer id. Note, the receiver is not required to use the provided public key.
+// Also note the signature is over the whole marshalled headmsg instead of just
+// a cidStr. This protects us from forgetting to sign something in the future if
+// we add to HeadMsg.
+type SignedHeadMsgEnvelope struct {
```

Comment:
I'm thinking we should add json struct tags to rename this to be more json-like.